tools/yulPhaser/FitnessMetrics.cpp

```diff
@@ -50,3 +50,47 @@ size_t RelativeProgramSize::evaluate(Chromosome const& _chromosome)
 		static_cast<double>(optimisedSize) / unoptimisedSize * scalingFactor
 	));
 }
+
+size_t FitnessMetricAverage::evaluate(Chromosome const& _chromosome)
+{
+	assert(m_metrics.size() > 0);
+
+	size_t total = m_metrics[0]->evaluate(_chromosome);
```

Comment:
Why not start with zero?

Reply:
Initially I did but it was a tad easier to make changes in these four functions when I kept their structure identical and starting with zero won't work for `FitnessMetricMinimum`. It also makes it easier to spot the differences that actually matter.

Both approaches have their pros and cons but I did not dwell on it too much since it's a pretty trivial case. I would be fine with doing it either way.